feat✨: update OpenAI chat generation and add task details

edgardleal opened the pull request:
👾 **Summary**
- Update logic for generating chat messages using OpenAI API
- Include user instructions and Git commit diff in chat messages
- Enhance handling of task details if provided

🔗 **Details**
- Define struct `ChatMessage` with role and content fields
- Retrieve OpenAI token securely from environment variable
- Create messages vector with system and user messages
- Include user instructions and Git commit diff in chat messages
- Handle task details with Jira ticket information if provided

Closes #1

edgardleal committed:
feat✨: update OpenAI chat generation and add task details

👾 **Summary**
- Update logic for generating chat messages using OpenAI API
- Include user instructions and Git commit diff in chat messages
- Enhance handling of task details if provided

🔗 **Details**
- Define struct `ChatMessage` with role and content fields
- Retrieve OpenAI token securely from environment variable
- Create messages vector with system and user messages
- Include user instructions and Git commit diff in chat messages
- Handle task details with Jira ticket information if provided

codermarcos commented on src/providers/jira.rs:
```diff
@@ -15,7 +15,13 @@ pub struct JiraConfiguration {
+    if id == "skip" { // sory for this, lest think in a better strategy to deal with this case
+        return Ok(None);
+    }
     let url = format!(
```
What do you think about make it optional into the [`providers/mod.rs`](https://github.com/compiled-thoughts/git-ai/blob/0304680e18cf690afcbc6e7b818363e6b9d250d8/src/providers/mod.rs) intead of here?

codermarcos commented on src/providers/jira.rs:
```diff
@@ -34,28 +40,33 @@ pub async fn get_ticket(id: String, configuration: &JiraConfiguration) -> Result
 }
 
-pub fn get_fields(ticket: Value) -> Ticket {
+pub fn get_fields(ticket: Option<Value>) -> Ticket {
```
Following the previous suggestion you don't need to change this method!

codermarcos reviewed (approved):
Just a little change!
![TooSmallToRead-SmallGIF](https://github.com/compiled-thoughts/git-ai/assets/12430365/18589a2a-9191-4378-b3b7-48c67d2eebf5)

edgardleal committed:
chore(cli): 🚀 update direct CLI to handle optional ticket-id input

- Removed the requirement for the `ticket-id` argument in the `generate` subcommand
- Added a default value of an empty string for the `ticket-id` argument in the `run` function
